File: src/main/java/kingdoms/server/handlers/packets/client/CPacketSyncDataPlayer.java
package kingdoms.server.handlers.packets.client;

import kingdoms.api.network.AbstractPacket;
import kingdoms.server.PlayerProvider;
import net.minecraft.entity.player.EntityPlayer;

public final class CPacketSyncDataPlayer extends AbstractPacket
{
    public CPacketSyncDataPlayer(){}
    public CPacketSyncDataPlayer(PlayerProvider provider)
    {
        buf().writeInt(provider.getGoldTotal());
        buf().writeInt(provider.getBankGold());
        buf().writeInt(provider.getGlory());
    }

    @Override
    public void client(EntityPlayer player)
    {
        if (player != null && player.worldObj != null)
        {
            PlayerProvider.get(player).setGoldTotal(buf().readInt());
            PlayerProvider.get(player).setBankGold(buf().readInt());
            PlayerProvider.get(player).setGlory(buf().readInt());
        }
    }
}

File: src/main/java/kingdoms/server/handlers/packets/client/CPacketSyncDataPlayer.kt
package kingdoms.server.handlers.packets.client

import kingdoms.api.network.SimplePacket
import kingdoms.server.PlayerProvider
import net.minecraft.entity.player.EntityPlayer

class CPacketSyncDataPlayer(): SimplePacket()
{
    constructor(provider: PlayerProvider): this()
    {
        buf().writeInt(provider.goldTotal)
        buf().writeInt(provider.bankGold)
        buf().writeInt(provider.glory)
        buf().writeBoolean(provider.hunter)
    }

    override fun client(player: EntityPlayer)
    {
        if (player.worldObj != null)
        {
            val provider = PlayerProvider.get(player)
            provider.goldTotal = buf().readInt()
            provider.bankGold = buf().readInt()
            provider.glory = buf().readInt()
            provider.hunter = buf().readBoolean()
        }
    }
}